### DISCUSSION TITLE
Any way to have `@variable` be a separate group if it's inside a `@string`?

### DISCUSSION THREAD

**Discussion by tuurep** [nvim-treesitter/nvim-treesitter#5964] [Q&A]
Hi

I have a goal to have variable names be default fg color, *unless* they are inside a string, they'd have another highlight group

![image](https://github.com/nvim-treesitter/nvim-treesitter/assets/15036238/2f92b5f8-eded-4d3c-a263-5a1122b4736b)

to be exact, have `my_var` inside string be cyan.

Is there any way to use the fact that the `my_var` inside the string matches both `@string` and `@variable` at the same time, to create a new capture/highlight group?

* `:Inspect` on `my_var`:

![image](https://github.com/nvim-treesitter/nvim-treesitter/assets/15036238/35654dde-a82d-439f-8f05-3be57daee74e)

From what I've seen it seems like it's possible to write queries per language to create a new capture group, but would require to have a `highlight.scm` for each language separately (please correct me if I'm wrong).

**Comment by amaanq** [comment]
yeah you can write a query that targets variables inside strings, and sets a new capture (highlight group) for that. take a look at the bash highlights query and `:h treesitter-query-modeline` to extend your queries.
